metrics/web/docs/03_widget_structure_organization.md

```diff
@@ -1,54 +1,350 @@
+
+Let's consider the pros & cons of creating a separate widget for each metric: 
+
+Pros: 
+- SRP: one widget for one view.
+- It makes the namings a bit more intuitive.
+- Reduces the number of changes to make one widget looks different.
+
+Cons: 
+- Increases code duplication in implementation and tests.
+- Increases the amount of code.
+
+Another way is to create a single configurable high-level widget for displaying these metrics. Let's consider the pros & cons of this approach:
+
+Pros: 
+- Keeps your code DRY.
+- It does not violate the single responsibility principle.
+- Reduces the amount of code and thus reduces the number of possible bugs, errors, etc.
+
+Cons: 
+- Makes your code less maintainable.
+- Increases amount of changes to make one widget looks different.
+
+So, it seems to be better to create a separate widget for each view even if these widgets look identical currently. It will allow us to simply change one of them later and increase maintainability.
 
 ## Metrics Theme guidelines
+
+### Metrics Theme structure [Pending Approval]
 > Explain and diagram the metrics theme structure.
 
+In order to support the application's appearance in a clear and maintainable way, the Metrics Web Application uses themes. The theme is a configuration for the appearance (colors, text style, etc.) of a part of UI. Themes allow configuring how the UI or its part looks like and introduces the great support to change this appearance reactively (e.g. dark and light themes).
+
+Our main concept of the theme is that we have a `MetricsThemeData` class that contains all the theme data for widgets. We can obtain the `MetricsThemeData` using the `MetricsTheme.of(context)` method. The `MetricsThemeData` consists of classes that implement the  `AttentionLevelThemeData` interface. This interface provides an `attentionLevel` field that provides the different variations of styles, applicable for separate group of widgets. Let's consider the class diagram that represents structure of `MetricsThemeData` and the relationships between classes in the theme data and widgets: 
+
+![Metrics Theme Structure Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/software-platform/monorepo/widget_stucture_organization_document/metrics/web/docs/diagrams/theme_data_class_diagram.puml)
+
+The main idea of the Metrics Theme inspired by Flutter default MaterialTheme that provides the `InheritedWidget` with the theme data to widgets. So, we have a `MetricsThemeData` class that contains the theme data for all the Metric widgets. Also, we have a `MetricsTheme` widget - the `InheritedWidget` that provides the `MetricsThemeData` to descendant widgets. To simplify the mechanism of changing the theme (from light to dart), there is the `MetricsThemeBuilder` widget that holds the light and dark themes and builds the MetricsTheme widget depending on the current theme state.
```

Comment:
If feels like this should be moved closer to the top of the section as it provides great intro into how themes work.